tech(web&admin): Create a Password component

meardagoat opened the pull request:
## Description

This PR creates a PasswordComponent.vue in both web and admin directories.

## Changes

- Add PasswordComponent with toggle visibility functionality
- Component accepts label prop (default: 'password')
- Input switches between password and text type
- Created in both web and admin directories

## Related Issue

Closes #81

theotime2005 reviewed (changes requested):
You must test your component

theotime2005 commented on admin/src/components/PasswordComponent.vue:
```diff
@@ -0,0 +1,96 @@
+defineProps({
+  label: {
+    type: String,
+    default: "password",
```
We are in french so is "mot de passe"

theotime2005 commented on admin/src/components/PasswordComponent.vue:
```diff
@@ -0,0 +1,96 @@
+  modelValue: {
+    type: String,
+    default: "",
+  },
```
I don't understand why the value is defined in props, it's a variable no ?

theotime2005 commented on admin/src/components/PasswordComponent.vue:
```diff
@@ -0,0 +1,96 @@
+        type="button"
+        @click="togglePasswordVisibility"
+        class="toggle-button"
+        :aria-label="showPassword ? 'Hide password' : 'Show password'"
```
We are in french:)
```suggestion
        :aria-label="showPassword ? 'Masquer le mot de passe' : 'Afficher le mot de passe'"
```

theotime2005 reviewed (changes requested):
The globality is so correct, but you must considere suggestions. Don't forget to apply on web also.

theotime2005 commented on admin/tests/integration/components/PasswordComponent.spec.js:
```diff
@@ -0,0 +1,123 @@
+
+    // then
+    expect(input.attributes("type")).toBe("text");
+
```
You can directly update variable of showPassword with this line
```suggestion
    wrapper.vm.showPassword = true;

```

theotime2005 reviewed (commented):
It's so good, but I continuing to think that modelValue mustn't being a prop. Password input will bee always an empty field. What do you think?

meardagoat committed:
tech(web): Create PasswordComponent with tests

- Add PasswordComponent with toggle visibility functionality
- Component accepts label prop (default: 'mot de passe')
- Input switches between password and text type
- Add modelValue prop and update:modelValue emit for v-model support
- Add French aria-labels for accessibility
- Add comprehensive integration tests
- Apply review suggestions (French labels, test structure improvements)

Closes #81

meardagoat committed:
tech(admin): Create PasswordComponent with tests

- Add PasswordComponent with toggle visibility functionality
- Component accepts label prop (default: 'mot de passe')
- Input switches between password and text type
- Add modelValue prop and update:modelValue emit for v-model support
- Add French aria-labels for accessibility
- Add comprehensive integration tests
- Apply review suggestions (French labels, test structure improvements)

Closes #81

theotime2005 reviewed (approved):
Reviewed, maby we will adjust component when will used. Let's go :rocket:

edp5-service commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/edp5/kompagnon/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
